go/tools/gazelle/rules/generator.go

```diff
@@ -91,15 +94,25 @@ func (g *generator) Generate(rel string, pkg *build.Package) ([]*bzl.Rule, error
 		rules = append(rules, p)
 	}
 
-	libName := defaultLibName
-	r, err := g.generateLib(rel, libName, pkg)
+	cgoLibrary := ""
+	if len(pkg.CgoFiles) != 0 {
+		cgoLibrary = defaultCgoLibName
+		r, err := g.generateCgoCLib(rel, cgoLibrary, pkg)
+		if err != nil {
+			return nil, err
+		}
+		rules = append(rules, r)
+	}
```

Comment:
perhaps a warning also when CgoFiles is empty, but any of C/CXX/M files are not?